test/hpke_test.c

```diff
@@ -0,0 +1,1804 @@
+    g_pub_len = 10;
+    if (TEST_false(OSSL_HPKE_get_grease_value(testctx, NULL, NULL, &g_suite,
+                                              g_pub, &g_pub_len,
+                                              g_cipher, g_cipher_len)) != 1) {
+        overallresult = 0;
+    }
+    /* reset to work */
+    g_pub_len = OSSL_HPKE_TSTSIZE;
+    if (TEST_true(OSSL_HPKE_get_grease_value(testctx, NULL, NULL, &g_suite,
+                                             g_pub, &g_pub_len,
+                                             g_cipher, g_cipher_len)) != 1) {
+        overallresult = 0;
+    }
+    /* expansion */
+    expanded = OSSL_HPKE_get_ciphertext_size(g_suite, clearlen);
+    if (!TEST_size_t_gt(expanded,clearlen)) {
+        overallresult = 0;
+    }
+    enclen = OSSL_HPKE_get_public_encap_size(g_suite);
+    if (!TEST_size_t_ne(enclen,0))
+        overallresult = 0;
+    /* not really GREASE but we'll check ikmelen thing */
+    ikmelen = OSSL_HPKE_get_recommended_ikmelen(g_suite);
+    if (!TEST_size_t_ne(ikmelen,0))
+        overallresult = 0;
+
+    return overallresult;
+}
+
+/*
+ * Make a set of calls with odd parameters
+ */
+static int test_hpke_oddcalls(void)
+{
+    int erv = 0;
+    EVP_PKEY *privp = NULL;
+    unsigned char pub[OSSL_HPKE_TSTSIZE];
+    size_t publen = sizeof(pub);
+    int hpke_mode = OSSL_HPKE_MODE_BASE;
+    int bad_mode = 0xbad;
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    OSSL_HPKE_SUITE bad_suite = { 0xbad, 0xbad, 0xbad };
+    OSSL_HPKE_CTX *ctx = NULL;
+    OSSL_HPKE_CTX *rctx = NULL;
+    unsigned char plain[] = "quick brown fox";
+    size_t plainlen = sizeof(plain);
+    unsigned char enc[OSSL_HPKE_TSTSIZE];
+    size_t enclen = sizeof(enc);
+    unsigned char cipher[OSSL_HPKE_TSTSIZE];
+    size_t cipherlen = sizeof(cipher);
+    unsigned char clear[OSSL_HPKE_TSTSIZE];
+    size_t clearlen = sizeof(clear);
+    unsigned char fake_ikm[OSSL_HPKE_TSTSIZE];
+    OSSL_LIB_CTX *badctx = NULL;
+    char *badpropq = "yeah, this won't work";
+    uint64_t lseq = 0;
+    char giant_pskid[OSSL_HPKE_MAX_PARMLEN + 10];
+    unsigned char info[OSSL_HPKE_TSTSIZE];
+
+    /* many of the calls below are designed to get better test coverage */
+
+    /* NULL ctx calls */
+    OSSL_HPKE_CTX_free(NULL);
+    if (!TEST_false(OSSL_HPKE_CTX_set_seq(NULL, 1)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_get_seq(NULL, &lseq)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_authpub(NULL, pub, publen)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_authpriv(NULL, privp)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_ikme(NULL, NULL, 0)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_psk(NULL, NULL, NULL, 0)))
+        goto end;
+
+    /* make/break ctx */
+    if (!TEST_ptr(ctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                          testctx, "foo")))
+        goto end;
+    OSSL_HPKE_CTX_free(ctx);
```

Comment:
Set `ctx` to `NULL` after this - otherwise we'll get a double free if one of the later tests fail

Reply:
good catch, did that